diff --git a/src/com/google/javascript/jscomp/NameGenerator.java b/src/com/google/javascript/jscomp/NameGenerator.java
index ae381bf..6e1be9f 100644
--- a/src/com/google/javascript/jscomp/NameGenerator.java
+++ b/src/com/google/javascript/jscomp/NameGenerator.java
@@ -106,7 +106,8 @@ final class NameGenerator {
   NameGenerator(Set<String> reservedNames, String prefix,
       @Nullable char[] reservedCharacters) {
     this.reservedNames = reservedNames;
-    this.prefix = prefix;
+    //this.prefix = prefix;
+    this.prefix = "z$" + prefix;
 
     this.priorityLookupMap = Maps.newHashMapWithExpectedSize(
         NONFIRST_CHAR.length);
